// rebass/dist/emotion/Position.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Sticky = exports.Fixed = exports.Absolute = exports.Relative = exports.Position = undefined;

var _emotion = require('system-components/emotion');

var _emotion2 = _interopRequireDefault(_emotion);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Position = exports.Position = (0, _emotion2.default)('space', 'color', 'zIndex', 'top', 'right', 'bottom', 'left');

Position.displayName = 'Position';

var Relative = exports.Relative = (0, _emotion2.default)({
  is: Position
}, {
  position: 'relative'
});
Relative.displayName = 'Relative';

var Absolute = exports.Absolute = (0, _emotion2.default)({
  is: Position
}, {
  position: 'absolute'
});
Absolute.displayName = 'Absolute';

var Fixed = exports.Fixed = (0, _emotion2.default)({
  is: Position
}, {
  position: 'fixed'
});
Fixed.displayName = 'Fixed';

var Sticky = exports.Sticky = (0, _emotion2.default)({
  is: Position
}, {
  position: 'sticky'
});
Sticky.displayName = 'Sticky';

exports.default = Position;
