// rebass/dist/emotion/Progress.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Progress = undefined;

var _emotion = require('system-components/emotion');

var _emotion2 = _interopRequireDefault(_emotion);

var _styledSystem = require('styled-system');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Progress = exports.Progress = (0, _emotion2.default)({
  is: 'progress',
  width: 1,
  m: 0,
  px: 0,
  color: 'blue',
  bg: 'gray',
  borderRadius: 2
}, function (props) {
  return {
    boxSizing: 'border-box',
    diplay: 'block',
    height: '4px',
    overflow: 'hidden',
    appearance: 'none',
    '&::-webkit-progress-bar': {
      backgroundColor: (0, _styledSystem.theme)('colors.gray')(props)
    },
    '&::-webkit-progress-value': {
      backgroundColor: 'currentcolor'
    },
    '&::-moz-progress-bar': {
      backgroundColor: 'currentcolor'
    }
  };
});

Progress.displayName = 'Progress';

exports.default = Progress;
